### ISSUE TITLE
Allow shadow-cljs to be installed globally

### ISSUE THREAD

**Issue by kanaka** [thheller/shadow-cljs#135]
This commit added a check that forces shadow-cljs to be installed locally:
https://github.com/thheller/shadow-cljs/commit/8690ec1915cb487f8306e1b084468f767324888e

However, as a build tool, it's pretty inconvenient to be forced to have it installed locally.

Specifically, in my case I would like to have a Dockerfile for building ClojureScript libraries but being forced to have shadow-cljs installed in the project itself makes the Dockerfile messy and really slows down the development cycle.

Also, in a similar vein it would be nice to have a shadow-cljs command that does dependency download and AOT caching without requiring a shadow-cljs.edn file (so that don't have to be resolved everytime the docker image is used for a different project).

In other words, I would like to have a Dockerfile that looks like this:

```
FROM ubuntu 17.10

# nodejs and java install (elided)

RUN npm -g install shadow-cljs
shadow-cljs deps
```

I could then use the resulting image to quickly package up ClojureScript libraries with a single command (and no dependencies on the host except docker) like this:

```
docker run -v /cljs-lib:/cljs-lib -w /cljs-lib -it cljs-build shadow-cljs compile
```

**Comment by thheller** [comment]
I'm not quite sure I understand what you are trying to do? I don't know much about docker and avoid it whenever possible.

> dependency download and AOT caching without requiring a shadow-cljs.edn

How would it know what do download without a config specifying the `:dependencies`?

The local install is forced due to the interop with `babel`. The `shadow-cljs` package itself provides the interface and ensures that `babel-core` is installed. It got way too messy trying to make this work without knowing where to find the files.

Can you create a basic example I can play with and see what the problems are?

**Comment by thheller** [comment]
This is now sort of solved as a side effect.

`npm install shadow-cljs` will download and AOT compile the `shadow-cljs` release itself. It still needs to be installed in the project but the AOT compile is shared and only done once. Since the deps are downloaded into `~/.m2` they are also shared.

With a bit more rework the project local install might become optional again as well but maybe this already solves what you were trying to do?
